### PR TITLE
Now working on Julia 1.0

### PR DESCRIPTION
**Pull request by snthot**
I revised and fixed the code to be able to work on Julia 0.7 and 1.0

### CODE REVIEW

**Comment by chezou** [comment]
@snthot Thank you for sending PR with your great effort. I'll check it soon.

File: README.md [hunk 1]
```diff
@@ -1,12 +1,18 @@
 # MeCab.jl
 
-[![Build Status](https://travis-ci.org/chezou/MeCab.jl.svg?branch=master)](https://travis-ci.org/chezou/MeCab.jl)
-[![Coverage Status](https://coveralls.io/repos/chezou/MeCab.jl/badge.svg?branch=master&service=github)](https://coveralls.io/github/chezou/MeCab.jl?branch=master)
-[![MeCab](http://pkg.julialang.org/badges/MeCab_0.3.svg)](http://pkg.julialang.org/?pkg=MeCab&ver=0.3)
-[![MeCab](http://pkg.julialang.org/badges/MeCab_0.4.svg)](http://pkg.julialang.org/?pkg=MeCab&ver=0.4)
+[![Build Status](https://travis-ci.org/snthot/MeCab.jl.svg?branch=master)](https://travis-ci.org/snthot/MeCab.jl)
```
Comment (chezou) [inline README.md]: Is there any reason to modify these lines?

I think "Build Status" badge should refer to `chezou/MeCab.jl` and package badges should be changed to 0.7 and 1.0. Coverage Status also should keep in the README.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 
+## Install
+
+I am not sure if the owner will merge my pull request. However, now, you can install this fork directly to enjoy using the package on Julia 0.7 and 1.0.
+In Julia REPL press ']' to enter package manager. Then print the following:
+
+```
+add https://github.com/snthot/MeCab.jl.git
+```
+
```
Comment (chezou) [inline README.md]: This part should be removed.

File: README.md [hunk 3]
```diff
@@ -70,6 +76,8 @@ julia> MeCab.libmecab
 ```
 
 ## Credits
+September 2018, modified for Julia 1.0 by snthot
+
```
Comment (chezou) [inline README.md]: After merging it, I will add "contributors" section on README, so could you remove it?

File: README.md (hunk 1 above)
Comment (snthot) [reply README.md]: Oh I see,let me try to fix

Comment (snthot) [reply README.md]: I returned to the original text, 
added badges 0.7 and 1.0 but It seems doesn't work now.
Please feel free to modified.

File: README.md (hunk 2 above)
Comment (snthot) [reply README.md]: yes, removed

File: README.md (hunk 3 above)
Comment (snthot) [reply README.md]: well, removed.
I'm new, please forgive me.

File: README.md [hunk 4]
```diff
@@ -4,9 +4,14 @@
...
 [![MeCab](http://pkg.julialang.org/badges/MeCab_0.3.svg)](http://pkg.julialang.org/?pkg=MeCab&ver=0.3)
 [![MeCab](http://pkg.julialang.org/badges/MeCab_0.4.svg)](http://pkg.julialang.org/?pkg=MeCab&ver=0.4)
+[![MeCab](http://pkg.julialang.org/badges/MeCab_0.7.svg)](http://pkg.julialang.org/?pkg=MeCab&ver=0.7)
+[![MeCab](http://pkg.julialang.org/badges/MeCab_1.0.svg)](http://pkg.julialang.org/?pkg=MeCab&ver=1.0)
+
+
 
 Julia bindings for Japanese morphological analyzer [MeCab](http://mecab.googlecode.com/svn/trunk/mecab/doc/index.html)
 
+
```
Comment (chezou) [inline README.md]: [nits] Don't need to add this blank line

File: README.md [hunk 5]
```diff
@@ -4,9 +4,14 @@
 [![Coverage Status](https://coveralls.io/repos/chezou/MeCab.jl/badge.svg?branch=master&service=github)](https://coveralls.io/github/chezou/MeCab.jl?branch=master)
 [![MeCab](http://pkg.julialang.org/badges/MeCab_0.3.svg)](http://pkg.julialang.org/?pkg=MeCab&ver=0.3)
 [![MeCab](http://pkg.julialang.org/badges/MeCab_0.4.svg)](http://pkg.julialang.org/?pkg=MeCab&ver=0.4)
```
Comment (chezou) [inline README.md]: Please remove old julia badges

File: README.md (hunk 1 above)
Comment (chezou) [reply README.md]: No worries. It could be resolved after publishing this package.

File: README.md (hunk 5 above)
Comment (snthot) [reply README.md]: sure, removed badges for 0.3, 0.4

**Comment by chezou** [comment]
Thanks for your PR @snthot ! I've just merged.

**Comment by snthot** [comment]
Wonderful! Last but not least. I just found that the following function should be rewrite to this:
function create_surface(raw::MecabRawNode)
  local surface::String
  surface = try
     unsafe_string(raw.surface, raw.length)
  catch
     unsafe_string(raw.surface)
  end
end
this might run faster.

**Comment by chezou** [comment]
@snthot oh, could you send another PR for it?

**Comment by snthot** [comment]
sure!

**Pull request merged**